giraph-core/src/main/java/org/apache/giraph/comm/aggregators/CountingOutputStream.java

```diff
@@ -36,7 +36,7 @@
    * Default constructor
    */
   public CountingOutputStream() {
-    dataOutput = new ExtendedByteArrayDataOutput();
+    dataOutput = new UnsafeByteArrayOutputStream();
```

Comment:
These changes allow us to write to the underlying buffer without having to first write to Chunked stream. However, what we write to stream is different depending on if the temporary Chunked streams is used, since the chunk size is also written. So we need to be consistent between write and read for kryo serialization. If the temporary chunk is not used when writing, casting the byte stream to ordinary ByteArrayInputStream when reading,  would cause error as we assume it used chunked stream. Since we know all our streams derives from UnsafeByteArray[Input/Outut]Stream, we can safely cast the byte array to unsafeByteArray stream.